modules/custom/activity_viewer/src/Plugin/views/row/ActivityRow.php

```diff
@@ -30,8 +31,32 @@ public function preRender($result) {
 
       foreach ($result as $row) {
         $render_result = [];
-        $render_result[] = $row;
         $entity = $row->_entity;
+        $target_type = $entity->get('field_activity_entity')->getValue()[0]["target_type"];
+
+        if ($target_type === "comment") {
+          $comment_id = $entity->get('field_activity_entity')->getValue()[0]["target_id"];
+          $query = \Drupal::database()->select('comment_field_data', 'c');
+          $query->addField('c', 'entity_id');
+          $query->condition('c.cid', $comment_id);
+          $post_id = $query->execute()->fetchField();
+
+          $query = \Drupal::database()->select('post_field_data', 'p');
+          $query->fields('p');
+          $query->condition('p.id', $post_id);
+          $post = $query->execute()->fetchAll();
+
+          $current_user_id = \Drupal::currentUser()->id();
+          $current_user = User::load($current_user_id);
+
+          if ($post->status =! "0" || $post->user_id === $current_user_id || $current_user->hasRole('administrator') ) {
```

Comment:
- `$post->status !== 0` make it strict.
- `$post->user_id === $current_user_id` users that create the post don't necessarily have access to the comments if an SM / CM has unpublished the post I believe. Right?
- `$current_user->hasRole('administrator')` We should check for a permission, SM / CM are also able to see unpublished items.